src/mpid/ch4/netmod/ofi/func_table.c

```diff
@@ -18,134 +18,140 @@ MPL_SUPPRESS_OSX_HAS_NO_SYMBOLS_WARNING;
 #include <mpidimpl.h>
 #include "netmod_direct.h"
 MPIDI_NM_funcs_t MPIDI_NM_ofi_funcs = {
-    MPIDI_NM_mpi_init_hook,
-    MPIDI_NM_mpi_finalize_hook,
-    MPIDI_NM_progress,
-    MPIDI_NM_mpi_comm_connect,
-    MPIDI_NM_mpi_comm_disconnect,
-    MPIDI_NM_mpi_open_port,
-    MPIDI_NM_mpi_close_port,
-    MPIDI_NM_mpi_comm_accept,
-    MPIDI_NM_comm_get_lpid,
-    MPIDI_NM_get_local_upids,
-    MPIDI_NM_upids_to_lupids,
-    MPIDI_NM_create_intercomm_from_lpids,
-    MPIDI_NM_mpi_comm_create_hook,
-    MPIDI_NM_mpi_comm_free_hook,
-    MPIDI_NM_am_request_init,
-    MPIDI_NM_am_request_finalize,
-    MPIDI_NM_am_send_hdr,
-    MPIDI_NM_am_isend,
-    MPIDI_NM_am_isendv,
-    MPIDI_NM_am_send_hdr_reply,
-    MPIDI_NM_am_isend_reply,
-    MPIDI_NM_am_hdr_max_sz,
-    MPIDI_NM_am_recv
+    .mpi_init                    = MPIDI_NM_mpi_init_hook,
+    .mpi_finalize                = MPIDI_NM_mpi_finalize_hook,
+    .progress                    = MPIDI_NM_progress,
+    .mpi_comm_connect            = MPIDI_NM_mpi_comm_connect,
+    .mpi_comm_disconnect         = MPIDI_NM_mpi_comm_disconnect,
+    .mpi_open_port               = MPIDI_NM_mpi_open_port,
+    .mpi_close_port              = MPIDI_NM_mpi_close_port,
+    .mpi_comm_accept             = MPIDI_NM_mpi_comm_accept,
+    /* Routines that handle addressing */
+    .comm_get_lpid               = MPIDI_NM_comm_get_lpid,
+    .get_local_upids             = MPIDI_NM_get_local_upids,
+    .upids_to_lupids             = MPIDI_NM_upids_to_lupids,
+    .create_intercomm_from_lpids = MPIDI_NM_create_intercomm_from_lpids,
+    .mpi_comm_create_hook        = MPIDI_NM_mpi_comm_create_hook,
+    .mpi_comm_free_hook          = MPIDI_NM_mpi_comm_free_hook,
+    /* Request allocation routines */
```

Comment:
Sounds a bit confusing, because as far as I understand `am_request_init` and `am_request_finalize` do not perform allocation/free. Maybe just "Request initialization/cleanup routines" is better.